calyx-py/calyx/builder_util.py

```diff
@@ -0,0 +1,175 @@
+# pylint: disable=import-error
+import calyx.builder as cb
+
+
+def insert_eq(comp: cb.ComponentBuilder, left, right, cellname, width):
+    """Inserts wiring into component {comp} to check if {left} == {right}.
+    1. Within {comp}, creates a combinational group called {cellname}_group.
+    2. Within the group, creates a cell {cellname} that checks equalities of {width}.
+    3. Puts the values {left} and {right} into the cell.
+    4. Returns the equality-checking cell and the overall group.
+    """
+    eq_cell = comp.eq(cellname, width)
+    with comp.comb_group(f"{cellname}_group") as eq_group:
+        eq_cell.left = left
+        eq_cell.right = right
+    return eq_cell, eq_group
+
+
+def insert_lt(comp: cb.ComponentBuilder, left, right, cellname, width):
+    """Inserts wiring into component {comp} to check if {left} < {right}.
+    1. Within {comp}, creates a combinational group called {cellname}_group.
+    2. Within the group, creates a cell {cellname} that checks less-than of {width}.
+    3. Puts the values {left} and {right} into the cell.
+    4. Returns the less-than-checking cell and the overall group.
+    """
+    lt_cell = comp.lt(cellname, width)
+    with comp.comb_group(f"{cellname}_group") as lt_group:
+        lt_cell.left = left
+        lt_cell.right = right
+    return lt_cell, lt_group
+
+
+def insert_add(comp: cb.ComponentBuilder, left, right, cellname, width):
+    """Inserts wiring into component {comp} to compute {left} + {right}.
+    1. Within {comp}, creates a combinational group called {cellname}_group.
+    2. Within the group, creates a cell {cellname} that computes sums of {width}.
+    3. Puts the values {left} and {right} into the cell.
+    4. Returns the summing cell and the overall group.
+    """
+    add_cell = comp.add(cellname, width)
+    with comp.comb_group(f"{cellname}_group") as add_group:
+        add_cell.left = left
+        add_cell.right = right
+    return add_cell, add_group
+
+
+def insert_sub(comp: cb.ComponentBuilder, left, right, cellname, width):
+    """Inserts wiring into component {comp} to compute {left} - {right}.
+    1. Within {comp}, creates a combinational group called {cellname}_group.
+    2. Within the group, creates a cell {cellname} that computes differences of {width}.
+    3. Puts the values {left} and {right} into the cell.
+    4. Returns the subtracting cell and the overall group.
+    """
+    sub_cell = comp.sub(cellname, width)
+    with comp.comb_group(f"{cellname}_group") as sub_group:
+        sub_cell.left = left
+        sub_cell.right = right
+    return sub_cell, sub_group
+
+
+def insert_incr(comp: cb.ComponentBuilder, reg, cellname, group):
```

Comment:
It might be worth considering making this take an arbitrary constant value to add, which defaults to 1.